src/utils/featheryClient/index.ts

```diff
@@ -474,6 +528,23 @@ export default class FeatheryClient extends IntegrationClient {
     );
   }
 
+  async submitCustom(customKeyValues: { [key: string]: any }, override = true) {
+    if (this.draft || this.noSave) return;
+    const shouldFlushPendingUpdates = this.formContext?.isLastStep();
```

Comment:
IMO instead of doing the check here, you can have a separate boolean parameter that triggers the flush, and you can trigger the flush each time submitStep or registerEvent (when the event type is a step completion) is called. That will be more robust and guarantee we get the data the user intends to set before critical event times (e.g. integration runs) while also solving the problem of a ton of submitCusotm calls stacked on each other